packages/web-extension/test/walletManager/util.test.ts

```diff
@@ -0,0 +1,29 @@
+import { AsyncKeyAgent } from '@cardano-sdk/key-management';
+import { Cardano } from '@cardano-sdk/core';
+import { getWalletId } from '../../src';
+
+describe('getWalletId', () => {
+  const pubKey = Cardano.Bip32PublicKey(
+    // eslint-disable-next-line max-len
+    '3e33018e8293d319ef5b3ac72366dd28006bd315b715f7e7cfcbd3004129b80d3e33018e8293d319ef5b3ac72366dd28006bd315b715f7e7cfcbd3004129b80d'
+  );
+  const chainId: Cardano.ChainId = {
+    networkId: Cardano.NetworkId.testnet,
+    networkMagic: Cardano.NetworkMagics.Preview
+  };
+  const mockKeyAgent = {
+    extendedAccountPublicKey: async () => Promise.resolve(pubKey),
+    getChainId: async () => Promise.resolve(chainId)
+  } as AsyncKeyAgent;
+
+  // Avoid using the same hashing logic as production.
+  // Used `b2sum` to get blake2b hash from pubKey.
+  const keyHashFromB2Sum =
+    // eslint-disable-next-line max-len
+    '1e73bcd61983f86626359b11a437bfc5e1a85ca3db90c79ea02f93dab150ad3f2ddf2e2f1c123b5d353827576a2a758487a6df59fcc47e17a28bfc62fd6098eb';
+
+  it('creates unique walletId from chainId and root public key hash', async () => {
+    const expectedWalletId = `${chainId.networkId}-${chainId.networkMagic}-${keyHashFromB2Sum}`;
+    expect(await getWalletId(mockKeyAgent)).toEqual(expectedWalletId);
```

Comment:
I don't think this test has much value as it's very tightly coupled with implementation (basically duplicates the code, just tests the hashing algorithm used). How about calling the function with different combinations of networkId/networkMagic/pubKey and expecting the returned strings to be unique?

Reply:
Good point. Updated tests to verify that walletId is unique when any of the networkId/networkMagic/publicKey changes: [7fb0dd5](https://github.com/input-output-hk/cardano-js-sdk/pull/560/commits/7fb0dd528e1c2d3b9d0eb4556d3f7e35ffe08f48)